### DISCUSSION TITLE
`do_orm_execute` create a completely new result

### DISCUSSION THREAD

**Discussion by jvllmr** [sqlalchemy/sqlalchemy#10804] [Usage Questions] [answered]
The dogpile.cahing result shows how one can modify query results, but the example caches complete results for re-use.

That brings me to the question: How can I create a totally new Result object? For instance a result created with already persisted, mapped objects I get from somewhere else.

**Comment by zzzeek** [comment, marked as answer]
oh like to construct a fake result from objects....try out [IteratorResult](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/engine/result.py#L2220) and likely send it [SimpleResultMetaData](https://github.com/sqlalchemy/sqlalchemy/blob/02472e8b65ac4062f2c3e7cee19608c801fba14c/lib/sqlalchemy/engine/result.py#L250).

a test fixture that does this can be seen at https://github.com/sqlalchemy/sqlalchemy/blob/02472e8b65ac4062f2c3e7cee19608c801fba14c/test/base/test_result.py#L271

**Reply by jvllmr** [reply]
Thanks for the hint! It works. However, when I use the Query API the query yields tuples instead of the objects.

**Reply by jvllmr** [reply]
For Query API I did:

```python
def is_query_api(orm_execute_state: ORMExecuteState) -> bool:
    return getattr(
        orm_execute_state.execution_options.get("_sa_orm_load_options", object()),
        "_legacy_uniquing",
        False,
    )


class QueryAPIIteratorResult(IteratorResult):
    @property
    def _row_getter(self):
        return None
```

This might not be optimal for edge cases, but it works for simple queries.

**Reply by zzzeek** [reply]
to make it work with the query API you would need to set some parameters on the result, take a look at how https://github.com/sqlalchemy/sqlalchemy/blob/94d20a225b4806f89a399da4704724284356598c/lib/sqlalchemy/orm/query.py#L2849 works
